packages/Search/src/details/DTK_DetailsTreeTraversal.hpp

```diff
@@ -66,7 +66,10 @@ struct TreeTraversal
     KOKKOS_INLINE_FUNCTION
     static Node const *getRoot( BVH<DeviceType> bvh )
     {
-        return bvh._internal_nodes.data();
+        if ( bvh._internal_nodes.extent( 0 ) > 0 )
+            return bvh._internal_nodes.data();
+        else
+            return bvh._leaf_nodes.data();
```

Comment:
`TreeTraversal::getRoot()` is not meant to be exposed to the user.  That's the main reason why I did not expose directly it in `BVH`.  It is only used in `spatial_query()` and `nearest_query()` and it would give a segfault if a Node pointer it returns was invalid.